src/global_data/4C_global_data_read.cpp

```diff
@@ -1477,6 +1478,9 @@ namespace
         {
           const auto& set = node_set.point_ids;
           node_sets[id] = std::vector<int>(set.begin(), set.end());
+
+          // 4C relies on a sorted node-set
+          std::ranges::sort(node_sets[id]);
```

Comment:
Seeing this, I definitely think the sorting should move into the node set itself rather than doing it here.